### DISCUSSION TITLE
Guidance Needed for Using DemoMars_fullRT_1D_clearsky.arts in ARTS Workflow

### DISCUSSION THREAD

**Discussion by jayeshgit65** [atmtools/arts#887] [Q&A] [answered]
Hi everyone,

I’m currently working with the DemoMars_fullRT_1D_clearsky.arts file from arts-2.4.0/controlfiles/planetary_toolbox/demos/passive (created by @Jana Mendrok). While I’m familiar with working on ARTS main control files (setting up workspaces, defining variables, etc.), I’m unsure how to properly invoke or integrate this specific demo file into my workflow.

Here are my specific concerns:

**How to Call the Demo File:**
How can I include and execute the DemoMars_fullRT_1D_clearsky.arts file in my main ARTS control script? For example, should I directly include the file as shown below, or is there a better practice?

INCLUDE "general/general.arts"
INCLUDE "general/continua.arts"
INCLUDE "general/agendas.arts"
INCLUDE "general/planet_mars.arts"
INCLUDE "planetary_toolbox/demos/passive/DemoMars_fullRT_1D_clearsky.arts"

**Integration with Python (pyarts):**
What is the recommended way to use this demo in a Python-based workflow, particularly with pyarts in a Jupyter Notebook? I want to efficiently call and run the demo file within my Python code.

**Compatibility Considerations:**
I’m currently using pyarts 2.4 for compatibility reasons but have prior experience with newer versions. Is there anything specific I should be aware of when working with this older version in offline mode?

I’ve reviewed the ARTS user guide but couldn’t find detailed instructions on integrating these demo files. Any tips, examples, or best practices would be greatly appreciated!

Thank you in advance for your support. 

[DemoMars_fullRT_1D_clearsky.arts.txt](https://github.com/user-attachments/files/17981842/DemoMars_fullRT_1D_clearsky.arts.txt)

**Comment by riclarsson** [comment, marked as answer]
Dear @JQYESH422x ,

You tagged a person that is not related to our project.  I do not expect you will get answers from them.

I would not expect great integration with python in ARTS 2.4.  There is some code for it, but it mostly started working pythonically in 2.5 and later.  Your easiest way of integrating with python should thus be that you just run the ARTS controlfiles standalone and store XML output files.  You then load these files into python using the XML parts of pyarts.

If you still want to try, ARTS 2.4 should come with a translation code available in https://github.com/atmtools/arts/tree/v2.4.x/python/bin.  But please take care about using this code as I am not sure for what version it stopped deleting the original .arts file.

About the planetary toolbox and using controlfiles, in my experience they generally produced a readable error message if they fail to parse.  If they did parse but failed for other reasons, these messages were also generally OK to follow.  So the only answer I can give is that you follow the instructions of the error messages that you receive until you can run all code.  Then modify the code to compute your setup.

**Reply by jayeshgit65** [reply]
Thank you for your guidance @riclarsson .
